src/main/java/org/springframework/data/jpa/repository/query/StoredProcedureAttributeSource.java

```diff
@@ -100,36 +101,28 @@ private String deriveProcedureNameFrom(Method method, Procedure procedure) {
 	private StoredProcedureAttributes newProcedureAttributesFrom(Method method,
 			NamedStoredProcedureQuery namedStoredProc, Procedure procedure) {
 
-		String outputParameterName = null;
-		Class<?> outputParameterType = null;
+		List<String> outputParameterNames = new ArrayList<>();
+		List<Class<?>> outputParameterTypes = new ArrayList<>();
 
 		if (!procedure.outputParameterName().isEmpty()) {
-
 			// we give the output parameter definition from the @Procedure annotation precedence
-			outputParameterName = procedure.outputParameterName();
+			outputParameterNames.add(procedure.outputParameterName());
```

Comment:
How about allow multiple outputParamerterNames ?

Reply:
@minhnguyenvan95 multiple outputParameterNames are supported for `@NamedStoredProcedureQuery` see here for an example https://github.com/spring-projects/spring-data-jpa/blob/master/src/test/java/org/springframework/data/jpa/domain/sample/User.java#L77